src/comestible_inv_listitem.cpp

```diff
@@ -0,0 +1,771 @@
+                    from_vehicle )
+{
+    init( g->u.get_consumable_from( *list.front() ) );
+}
+
+void comestible_inv_listitem_food::init( const item &it )
+{
+    player &p = g->u;
+    islot_comestible edible = get_edible_comestible( p, it );
+
+    menu_color = it.color_in_inventory();
+
+    // statuses
+    const auto &comest = it.get_comestible();
+    cond.reserve( cond_size );
+    const bool eat_verb = it.has_flag( "USE_EAT_VERB" );
+    const bool is_food = eat_verb || comest->comesttype == "FOOD";
+    const bool is_drink = !eat_verb && comest->comesttype == "DRINK";
+
+    bool craft_only = false;
+    bool warm_only = false;
+
+    if( it.is_craft() ) {
+        craft_only = true;
+    } else if( is_food || is_drink ) {
+        for( const auto &elem : it.type->materials ) {
+            if( !elem->edible() ) {
+                craft_only = true;
+                break;
+            }
+        }
+    }
+    if( !craft_only && it.item_tags.count( "FROZEN" ) && !it.has_flag( "EDIBLE_FROZEN" ) &&
+        !it.has_flag( "MELTS" ) ) {
+        warm_only = true;
+    }
+    input_context ctxt( "COMESTIBLE_INVENTORY" );
+    if( craft_only ) {
+        cond.emplace_back( ctxt.get_desc( "CRAFT_WITH" )[0], c_cyan );
+    } else if( warm_only ) {
+        cond.emplace_back( ctxt.get_desc( "HEAT_UP" )[0], c_cyan );
+    } else {
+        cond.emplace_back( ' ', c_cyan );
+    }
+
+    if( it.has_flag( "MUSHY" ) ) {
+        cond.emplace_back( 'm', c_yellow );
+    } else if( edible.stim > 0 ) {
+        cond.emplace_back( 's', c_yellow );
+    } else if( edible.stim < 0 ) {
+        cond.emplace_back( 'd', c_yellow );
+    } else {
+        cond.emplace_back( ' ', c_cyan );
+    }
+
+    //These numbers are arbitrary.
+    if( edible.addict > 7 ) {
+        if( edible.addict < 20 ) {
+            cond.emplace_back( 'a', c_yellow );
+        } else {
+            cond.emplace_back( 'a', c_red );
+        }
+    } else {
+        cond.emplace_back( ' ', c_cyan );
+    }
+
+    eat_error = p.can_eat( it );
+
+    exipres_in = "";
+    shelf_life = "";
+    if( edible.spoils > 0_turns ) {
+        if( !it.rotten() ) {
+            exipres_in = get_time_left_rounded( p, it );
+        }
+        shelf_life = to_string_clipped( edible.spoils );
+    }
+
+    calories = p.kcal_for( it );
+    quench = edible.quench;
+    joy = p.fun_for( it ).first;
+}
+void comestible_inv_listitem_food::print_columns( std::vector<comestible_inv_columns> columns,
+        comestible_select_state selected_state, catacurses::window window, int right_bound,
+        int cur_print_y )
+{
+    // if it's ineddible, but cookable or reheatable we want to show full data
+    if( !eat_error.success() && eat_error.value() != edible_rating::INEDIBLE ) {
+        int max_x = right_bound;
+        is_selected = selected_state;
+        int w;
+        std::string print_str;
+        nc_color print_col;
```

Comment:
These should be defined inside the loop body. It also allows to make them `const` there.